module-6/lsp-example.js

```diff
@@ -0,0 +1,46 @@
+// LSP
+
+function SomeComponent() {
+  return (
+    <PictureDataProvider>
+      {pictureData => (
+        <>
+          <Topbar>
+            <ProfilePicture pictureData={pictureData} />
+          </Topbar>
+
+          <LeftBar>
+            <Avatar pictureData={pictureData} />
+          </LeftBar>
+
+          <Menu>
+            <NavigationPicture pictureData={pictureData} />
+          </Menu>
+        </>
+      )}
+    </PictureDataProvider>
+  )
+}
+
+ProfilePicture.propTypes = {
+  pictureData: PropTypes.shape({
+    url: PropTypes.string.isRequired,
+    size: PropTypes.number.isRequired,
+  })
+}
+Avatar.propTypes = {
+  pictureData: PropTypes.shape({
+    url: PropTypes.string.isRequired,
+    size: PropTypes.number.isRequired,
+  })
+}
+NavigationPicture.propTypes = {
+  pictureData: PropTypes.shape({
+    url: PropTypes.shape({ // <- WTF
```

Comment:
co WTF?

Reply:
The idea is that in the other components the `url` is a string, here it's an object. The simplest version of LSP tells that you shouldn't surprise people, here we are breaking the rule a little bit.